src/database.ts

```diff
@@ -21,6 +21,18 @@ export type Query<T extends Schema> = {
 	orderBy?: OrderBy<T>[];
 };
 
+export enum FieldType {
+	String = 1 << 0,
+	Number = 1 << 1,
+	Boolean = 1 << 4,
+	Null = 1 << 5,
+}
+
+// given a fieldType and the desired type, this function performs a bitwise operation to test membership
+export function containsType(fieldType: bigint, desiredType: FieldType) {
+	return (fieldType & BigInt(desiredType)) !== BigInt(0);
+}
+
```

Comment:
wondering if this is too extra. 

The reason why I wanted to add this is at first glance it may not be immediate that to check for validity of types we need to perform a bitwise membership test.